### PR TITLE
missions 2

### PR DESCRIPTION
**Pull request by TMTIME**
cleans up a few missions, gets cargo missions working

## About The Pull Request

fixes up a few missions, lets stations give out more than 1 mission. FUN

Cargo missions are now completed by using cargo torps. One is provided for each cargo mission.

Missions can now reward the players with items. For now, combat missions give the players a box of ammo

Screenshot of the missions UI with 3 active missions. ~~Ignore the first mission asking you to deliver cargo to itself. Thats because for testing I had Minsky spawn in Sol as well to speed stuff up~~ The UI will show all inbound and outbound hails, but for now I just wanted to get it working with missions.

idk what needs to happen to update SpacemanDMM but when that happens I'll gladly fix the typeless loops in this pr

## Why It's Good For The Game

its a feature that needed a little polish

Adds some spice and variety to the overmap gameplay.

## Changelog
:cl:
add: Cargo missions can now spawn
add: Missions can now reward the crew with items
add: Modular computer program to track active missions for the ship
tweak: Stations offer up to 5 missions now
fix: Notifications of players being given missions are now only shown to the crew of that ship.
/:cl:

### CODE REVIEW

File: nsv13/code/modules/overmap/traders.dm [hunk 1]
```diff
@@ -16,9 +16,19 @@
 	var/stock_delay = 0
 	var/image = "https://cdn.discordapp.com/attachments/701841640897380434/764534224291233822/unknown.png"
 	var/list/missions = list() //Missions
-	var/list/mission_types = list(/datum/nsv_mission/kill_ships) //Todo: fix up the cargo delivery objective and add it to this list.
+	var/list/possible_mission_types = list( // List of possible missions this trader may have with a % chance to spawn
```
Comment (TMTIME) [inline nsv13/code/modules/overmap/traders.dm:19]: doesnt work like that any more dumbas

File: nsv13/code/modules/overmap/traders.dm [hunk 2]
```diff
@@ -163,25 +183,23 @@
 /datum/trader/proc/give_mission(mob/living/user)
 	if(!isliving(user))
 		return
-	if(!mission_types.len || missions.len >= max_missions)
+		
+	var/list/valid_missions = list()
+	
+	for(var/m in missions) // Get all valid missions the crew qualifies for
+		var/datum/nsv_mission/mission = m
```
Comment (ike709) [inline nsv13/code/modules/overmap/traders.dm:190]:
```suggestion
	for(var/datum/nsv_mission/mission as() in missions) // Get all valid missions the crew qualifies for
```
Most efficient way to do a typeless for loop.

**Comment by TMTIME** [comment]
just need to look into delivering cargo before i undraft this.

File: nsv13/code/modules/overmap/traders.dm (hunk 2 above)
Comment (TMTIME) [reply nsv13/code/modules/overmap/traders.dm]: travis didnt like it f

Comment (ike709) [reply nsv13/code/modules/overmap/traders.dm:190]: Which is why I said you'd need to update SpacemanDMM.

**Review by Kmc2000 — APPROVED** [review]

**Pull request merged**